transmission_interface/include/transmission_interface/transmission.h

```diff
@@ -122,6 +126,20 @@ class Transmission
   virtual void actuatorToJointPosition(const ActuatorData& act_data,
                                              JointData&    jnt_data) = 0;
 
+  virtual void actuatorToJointAbsolutePosition(const ActuatorData&,
+                                             JointData&){
+    throw std::runtime_error("transmission does not support actuator to joint absolute position");
+  }
+
+  virtual void actuatorToJointTorqueSensor(const ActuatorData&,
+                                             JointData&){
+    throw std::runtime_error("transmission does not support actuator to joint torque sensor");
+  }
```

Comment:
These should be const, but I know that differs from all the other actuatorToJointX() and jointToActuatorX() functions. Is there a reason the existing functions aren't const?

Reply:
We should allow internal states to be used&altered if needed in these functions

Reply:
What internal state would/could be modified? Currently, the 3 transmissions implemented (differential, four bar, and simple) do not modify any internal state, and I can't think of an example where you'd want a data transformation function to have side effects.

Reply:
Transmissions are plugin-based, anyone can write their own with whatever internal logic they wish to use. Of course these are niche cases but my point is, there's a possibility of a use-case we shouldn't rule out proactively.

Reply:
Got it